dashboard/app/controllers/javabuilder_sessions_controller.rb

```diff
@@ -13,40 +13,105 @@ class JavabuilderSessionsController < ApplicationController
 
   # GET /javabuilder/access_token
   def get_access_token
-    teacher_list = get_teacher_list.join(',')
+    begin
+      require_default_params
+      params.require(:channelId)
+    rescue ActionController::ParameterMissing
+      return render status: :bad_request, json: {}
+    end
     channel_id = params[:channelId]
+
+    begin
```

Comment:
did you try having one helper method that just took a `override_sources` flag that each of these methods could call into? Would that help reduce duplicate code here?

Reply:
Yeah I thought about it, I think maybe if we're able to get rid of the storage_id stuff then it would look cleaner for them to use a shared helper. With the storage app logic, it seemed more appropriate to keep them separate from one another 🤷 , even if duplicated